api/replication.storage/v1alpha1/volumereplication_types.go

```diff
@@ -36,6 +36,24 @@ const (
 	ConditionValidated = "Validated"
 )
 
+// These are valid messages for various
+// conditions and states of Volume
+// replication.
+const (
+	MessagePromoted           = "volume is promoted to primary and replicating to secondary volume"
+	MessageHealthy            = "volume is healthy"
+	MessageNotResyncing       = "volume is not resyncing"
+	MessageValidated          = "volume is validated and met all prerequisites"
+	MessageFailedPromoted     = "failed to promote volume"
+	MessageFailedDemoted      = "failed to demote volume"
+	MessageFailedPreCondition = "failed to meet prerequisite"
+	MessageDemoted            = "volume is demoted to secondary"
+	MessageDegraded           = "volume is degraded"
+	MessageResyncTriggered    = "volume is resyncing changes from primary to secondary"
+	MessageResyncFailed       = "failed to resync"
+	MessageEnableVolumeFailed = "failed to enable volume replication"
+)
```

Comment:
These messages should also work fine for volume groups as well, Instead of Volume can we call it as `source` so that we don't need to change the messages if it's for volume or volume group.

Reply:
sounds good to me, will update

Reply:
I think lets take opinion from @nirs as well on how this messages are used in VGR, cause AFSIK VGR do add messages based on all the VR conditions.

Reply:
I think when using volume group replication (VGR) we will have one VR per PVC, so nothing should change in the VR. @ELENAGER do you want review this?

Reply:
As far as I understand, when using volume group replication, CSI is creating one VR per PVC internally, is that right? And ramen is reading status and all the info from VGR

Reply:
Nope, we are going to have single VR created for a volumegroupReplication as the datasource, its one per group in case of group replication

Reply:
ok, so the message should be:

- volumes: "failed to {action} volume: {details}"
- volume groups: "failed to {action} volume group: {details}"

Reply:
@yati1998 this is still not addressed yet? Is it going to be a followup, If yes please open an issue for tracking.